web/template/partial/stream/bookmarks.gohtml

```diff
@@ -0,0 +1,153 @@
+                        Add bookmark
+                    </button>
+                </div>
+                {{template "bookmark-list" .}}
+            </div>
+        </template>
+        <template x-if="showAddMenu">
+            <div class="p-4">
+                <div class="flex justify-between items-end mb-4">
+                    <h3 class="font-bold text-sm text-xl text-3">Bookmarks</h3>
+                    <button type="button" title="Close bookmark dialog"
+                            class="text-3 text-sm font-semibold hover:bg-gray-100 dark:hover:bg-gray-600 rounded-full px-2 py-1"
+                            @click="showAddMenu = !showAddMenu">
+                        <i class="fa-solid fa-undo mr-2"></i>
+                        <span>Go back</span>
+                    </button>
+                </div>
+                {{template "bookmark-dialog" .}}
+            </div>
+        </template>
+    </div>
+{{end}}
+
+{{/* 1st parameter: 'streamID' */}}
+{{define "bookmark-dialog"}}
+    <form x-data="{bookmark: new watch.BookmarkDialog({{.}})}"
+          x-init="bookmark.reset()"
+          @submit.prevent="bookmark.submit().then(() => {$dispatch('reloadbookmarks'); showAddMenu = false;})"
+          class="">
+        <div class="relative grid gap-5 md:gap-2 text-sm">
+            <div>
+                <label for="bookmark-description"
+                       class="block text-5">Description</label>
+                <input x-model="bookmark.request.Description"
+                       type="text" name="bookmark-description" id="bookmark-description"
+                       placeholder="Interesting..." autofocus="" required
+                       class="rounded px-4 py-3 mt-1 text-4 focus:outline-none border-0 bg-gray-50 w-full dark:bg-gray-600"/>
+            </div>
+            <div>
+                <label for="bookmark-timestamp" class="block text-5">Timestamp</label>
+                <div class="flex align-middle">
+                    <input x-model="bookmark.request.Hours"
+                           type="number" name="bookmark-timestamp-hour"
+                           id="bookmark-timestamp-hour"
+                           placeholder="" required
+                           class="rounded px-4 py-3 mt-1 text-4 focus:outline-none border-0 bg-gray-50 w-full dark:bg-gray-600"/>
+                    <span class="px-2 my-auto font-semibold text-5">:</span>
+                    <input x-model="bookmark.request.Minutes"
+                           type="number" name="bookmark-timestamp-minutes"
+                           id="bookmark-timestamp-minutes"
+                           placeholder="" required
+                           class="rounded px-4 py-3 mt-1 text-4 focus:outline-none border-0 bg-gray-50 w-full dark:bg-gray-600"/>
+                    <span class="px-2 my-auto font-semibold text-5">:</span>
+                    <input x-model="bookmark.request.Seconds"
```

Comment:
As far as I can tell, there is no (client-side) validation here, see e.g.
<img width="354" alt="grafik" src="https://user-images.githubusercontent.com/30267166/205436017-c7ddbe17-41e1-49a7-afb2-0b25b0fb01cf.png">
I think this is a good safety measure.

Reply:
Definitely. Great catch!